src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -54,16 +54,99 @@
     end
   end
 
-  describe 'GET #autocomplete_projectsindex' do
+  describe 'GET #autocomplete_projects' do
+    before do
+      create(:project, name: 'Apache')
+      create(:project, name: 'Apache2')
+      create(:project, name: 'openSUSE')
+      create(:maintenance_incident_project, name: 'ApacheMI')
+    end
+
+    context 'without search term' do
+      before do
+        get :autocomplete_projects
+        @json_response = JSON.parse(response.body)
+      end
+
+      it { expect(@json_response).to be_a(Array) }
+      it { expect(@json_response.length).to eq(3) }
+      it { expect(@json_response).to contain_exactly('Apache', 'Apache2', 'openSUSE') }
+      it { expect(@json_response).not_to include('ApacheMI') }
+    end
+
+    context 'with search term' do
+      before do
+        get :autocomplete_projects, term: 'Apache'
+        @json_response = JSON.parse(response.body)
+      end
+
+      it { expect(@json_response).to be_a(Array)}
+      it { expect(@json_response.length).to eq(2) }
+      it { expect(@json_response).to contain_exactly('Apache', 'Apache2') }
+      it { expect(@json_response).not_to include('ApacheMI') }
+      it { expect(@json_response).not_to include('openSUSE') }
```

Comment:
Same here. "it { expect(@json_response).to contain_exactly('Apache', 'Apache2') }" seems to be enough